src/web/handlers.rs

```diff
@@ -24,34 +31,31 @@ pub async fn get_tiles(
     trace!("get_tiles");
     metrics.incr("tiles.update");
 
-    let cinfo = request.connection_info();
     let settings = &state.settings;
-    let ip_addr_str = cinfo.remote_addr().unwrap_or({
-        let default = state
-            .adm_country_ip_map
-            .get("US")
-            .expect("Invalid ADM_COUNTRY_IP_MAP settting");
-        if let Some(country) = &treq.country {
-            state.adm_country_ip_map.get(country).unwrap_or(default)
-        } else {
-            default
+    let mut addr = None;
+    if let Some(header) = request.headers().get(&*X_FORWARDED_FOR) {
```

Comment:
Well, crap. https://docs.rs/actix-web/3.3.2/actix_web/dev/struct.ConnectionInfo.html#method.realip_remote_addr

Reply:
I based some of this code off of that method -- problem with it is its us of `Forwarded` makes it problematic (it's possibly spoofed unless nginx/LB are disallowing it)

Reply:
Docs on how it works: https://cloud.google.com/load-balancing/docs/https#x-forwarded-for_header

The way nginx parses the client IP out of the X-Forwarded-For header is something you'll want to copy - basically you put a bunch of trusted IPs, and the first untrusted IP it finds is what is considered the client IP address: http://nginx.org/en/docs/http/ngx_http_realip_module.html

Reply:
@jbuck, I think I remember a lot of this, and I think I had to deal with it with ChannelServer. I can't remember if we need to toss the trusted IPs or if they're trimmed before the header is handed off.

Reply:
They're not trimmed, here's an example from the logs: `"99.237.130.164, 34.96.87.143, 99.237.130.164"` (also there's a bug we need to fix where the the load balancer IP is repeated)

Reply:
Ah, right: https://github.com/mozilla-services/channelserver/blob/fc2c592f832740e340a5b901ed9c87d56d4ce015/channelserver/src/meta.rs#L166

Reply:
@jbuck Looks like we are not using the `Real IP` Nginx module for [Contile](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/topsites/k8s/charts/contile/templates/nginx-configmap.yaml). Shall we add those to the configuration file?

```
real_ip_header X-Forwarded-For;
set_real_ip_from 0.0.0.0/0;
# set_real_ip_from other_cdn_ips;
```

Reply:
👍  w/ more restrictive subnets than 0.0.0.0/0 and `real_ip_recursive on` (copy over topsites proxy's nginx config here, it looks good)